notifications.bs

```diff
@@ -926,15 +926,15 @@ result of the following steps:
     <li><p>Set <var>action</var>'s {{NotificationAction/icon}} to
     <var>entry</var>'s <a for=action>icon URL</a>.
 
-    <!-- XXX IDL dictionaries are usually returned by value, so don't need to be
-         immutable. But FrozenArray reifies the dictionaries to mutable JS
-         objects accessed by reference, so we explicitly freeze them.
-         It would be better to do this with IDL primitives instead of JS - see
-         https://www.w3.org/Bugs/Public/show_bug.cgi?id=29004 -->
-    <li><p>Call <a lt="freeze">Object.freeze</a> on <var>action</var>, to
-    prevent accidental mutation by scripts.
+    <li><p>Let <var>jsAction</var> be <var>action</var>, <a>converted to an ECMAScript value</a>.
 
-    <li><p>Append <var>action</var> to <var>frozenActions</var>.
+    <li><p>Perform <a abstract-op>SetIntegrityLevel</a>(<var>jsAction</var>, "frozen").
+
+    <li><p>Let <var>idlAction</var> be <var>jsAction</var>, <a>converted to an IDL value</a>.
```

Comment:
This isn't correct, at least as written. The conversion, presumably to an IDL `NotificationAction` type (although that isn't stated), will re-perform the Get()s of all the properties, which can be observable if Object.prototype is modified. Then we'll convert back to JS in the binding layer.

I don't think there's any way to do this rigorously as long as your return type is `FrozenArray<NotificationAction>` (instead of `FrozenArray<object>`). The conversion between JS objects and dictionaries is observable and side effecting.

Reply:
I cannot use `FrozenArray` then either since it's constructor takes an IDL value. I'm not sure I want to inline all of that. That would be rather bad and I'd likely make mistakes or it would result in mismatches down the line.

There is some mismatch here though with implementations I think where JS values are backed by IDL values and you can manipulate either. That's not conversion, but just grabbing the corresponding value, but IDL doesn't formalize that.